### PR TITLE
Make strapi compatible with knex v1

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Upgrade to knex 1

### Why is it needed?

Describe the issue you are solving.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

**Comment by Aurelsicoko** [comment]
I've done one hour of intense testing on all the SQLite packages (sqlite3, @vscode/sqlite3, and better-sqlite3) without even dropping the DB. Everything worked like a charm.

File: packages/core/database/package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "date-fns": "2.22.1",
     "debug": "4.3.1",
     "fs-extra": "10.0.0",
-    "knex": "0.95.6",
+    "knex": "^1.0.4",
```
Comment (petersg83) [inline packages/core/database/package.json:37]: Don't we prefer to fix the version?

File: packages/core/database/lib/schema/schema.js [hunk 2]
```diff
@@ -105,7 +105,7 @@ const getColumnType = attribute => {
     case 'increments': {
       return {
         type: 'increments',
-        args: [{ primary: true }],
+        args: [{ primary: true, primaryKey: true }],
```
Comment (petersg83) [inline packages/core/database/lib/schema/schema.js:108]: Are `primary` and `primaryKey` redundant? Can we remove `primary` ?

File: packages/core/database/lib/connection.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+const clientMap = {
+  'better-sqlite3': 'better-sqlite3',
+  '@vscode/sqlite3': 'sqlite',
+  sqlite3: LegacySqliteClient,
+};
+
+const getSqlitePackageName = () => {
+  // NOTE: allow forcing the package to use (mostly used for testing purposes)
+  if (typeof process.env.SQLITE_PKG !== undefined) {
+    return process.env.SQLITE_PKG;
```
Comment (petersg83) [inline packages/core/database/lib/connection.js:35]: Is it supposed to be documented?

File: packages/core/database/lib/connection.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  // NOTE: this tries to find the best sqlite module possible to use
+  // while keeping retro compatibiity
```
Comment (petersg83) [inline packages/core/database/lib/connection.js:39]: compatibility* missing the `l`

File: packages/core/database/lib/connection.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+const clientMap = {
+  'better-sqlite3': 'better-sqlite3',
+  '@vscode/sqlite3': 'sqlite',
+  sqlite3: LegacySqliteClient,
+};
+
+const getSqlitePackageName = () => {
+  // NOTE: allow forcing the package to use (mostly used for testing purposes)
+  if (typeof process.env.SQLITE_PKG !== undefined) {
```
Comment (petersg83) [inline packages/core/database/lib/connection.js:34]: I think you meant `'undefined'` :p

**Review by petersg83 — CHANGES_REQUESTED** [review]

File: packages/core/database/lib/schema/schema.js (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/database/lib/schema/schema.js:108]: Nope we can't it's for retro compatibility :)

File: packages/core/database/lib/connection.js (hunk 3 above)
Comment (alexandrebodin) [reply packages/core/database/lib/connection.js:35]: IMO it doesn't makes sense to document it in the public doc but open to it

File: packages/core/database/lib/connection.js (hunk 5 above)
Comment (alexandrebodin) [reply packages/core/database/lib/connection.js:34]: nice catch

File: packages/core/database/lib/connection.js (hunk 3 above)
Comment (petersg83) [reply packages/core/database/lib/connection.js:35]: Ok good. Me neither, just checking 👌

File: packages/core/database/lib/schema/schema.js (hunk 2 above)
Comment (petersg83) [reply packages/core/database/lib/schema/schema.js:108]: I see, didn't think of that :) Thank

File: packages/core/database/lib/connection.js (hunk 3 above)
Comment (derrickmehaffy) [reply packages/core/database/lib/connection.js:35]: We should still document it but can clarify it's mainly for testing.

**Comment by derrickmehaffy** [comment]
I'll review again later today, two things:

- Are we making better-sqlite3 the default
- Can we release a quick alpha either today or tomorrow so we can do some real world testing

**Review by derrickmehaffy — APPROVED** [review]
LGTM, did quite a bit of testing on the 3 main OSes:

Ubuntu 20.04 (AMD CPU) => `sqlite3`, `better-sqlite3`, MariaDB, and PostgreSQL
Windows 10 (AMD CPU) => `sqlite3`, `better-sqlite3`, MariaDB, and PostgreSQL
Mac (Intel CPU) => `sqlite3`, `better-sqlite3`, and PostgreSQL

I'll keep doing some testing on my arm Pi 4 but we do we need someone with an M1 Mac to test

**Comment by alexandrebodin** [comment]
> LGTM, did quite a bit of testing on the 3 main OSes:
> 
> Ubuntu 20.04 (AMD CPU) => `sqlite3`, `better-sqlite3`, MariaDB, and PostgreSQL Windows 10 (AMD CPU) => `sqlite3`, `better-sqlite3`, MariaDB, and PostgreSQL Mac (Intel CPU) => `sqlite3`, `better-sqlite3`, and PostgreSQL
> 
> I'll keep doing some testing on my arm Pi 4 but we do we need someone with an M1 Mac to test

I have built it on an M1 I can tell you it's working :)

**Review by petersg83 — APPROVED** [review]
LGTM, waiting for the tests

**Review by petersg83 — APPROVED** [review]
LGTM :)

**Pull request merged**